libs/libc/machine/Kconfig

```
@@ -137,6 +137,50 @@ config LIBC_ARCH_ELF_64BIT
 	default n
 	depends on LIBC_ARCH_ELF
 
+config LIBC_ARCH_ALLOW_KERNEL_ONLY
+	bool
+	default n
+
+config LIBC_ARCH_ALLOW_USER_ONLY
+	bool
+	default n
```

```suggestion
config LIBC_ARCH_ALLOW_KERNEL
	bool
	default n

config LIBC_ARCH_ALLOW_USER
	bool
	default n
```
The existence of a `LIBC_ARCH_BOTH` option contradicts the suffix `_ONLY` for the above configs.

---

Please refer to https://github.com/apache/nuttx/pull/9138#discussion_r1181085999

Using `LIBC_ARCH_ALLOW_xxx` without the `ONLY` suffix reminds me of something that should be selected by a chip to state where the arch-specific implementations would be allowed. This would require 1) the default value being y or 2) these options being selected by all the chips (currently, all the chips "allow" arch's libc in both the kernel and the userspace). 

The suffix `_ONLY` reminds me a different meaning, something like "ow, this arch has a different behavior than the usual. It allows the arch's libc to be used only by other kernel or by the userspace".

---

> The suffix _ONLY reminds me a different meaning, something like "ow, this arch has a different behavior than the usual. It allows the arch's libc to be used only by other kernel or by the userspace".

It is still contradictory. Having a `_ONLY` prefix implies a XOR logic.

---

I agree that `_ONLY` implies XOR logic. But there is a level between `_ALLOW_` and the choice itself: you can allow something that is XOR-ed when you have multiple choices. 

However, if It isn't that clear, let's find a more intuitive way of doing it.